src/utils.rs

```diff
@@ -241,38 +241,80 @@ pub fn get_encryption_key(enc_params: &EncryptionParams) -> Result<&[u8]> {
     }
 }
 
+struct DataBuffer {
+    data: *mut u8,
+    size: usize,
+}
+
 /// Copies a decrypted data into the [SecurityBufferType::Data] or [SecurityBufferType::Stream].
 ///
 /// If the provided buffers do not contain the [SecurityBufferType::Data] buffer, then it will try
 /// to write the data in the [SecurityBufferType::Stream]. Otherwise, the error will be returned.
 /// If the inner buffer is not large enough, then this function will return an error.
-pub fn save_decrypted_data(decrypted: &[u8], buffers: &mut [DecryptBuffer]) -> Result<()> {
-    let buffer = if let Ok(buffer) = DecryptBuffer::find_buffer_mut(buffers, SecurityBufferType::Data) {
-        if buffer.data().len() < decrypted.len() {
-            return Err(Error::new(
-                ErrorKind::DecryptFailure,
-                "Decrypted data can not be larger then encrypted one.",
-            ));
-        }
-        buffer
-    } else {
-        let buffer = DecryptBuffer::find_buffer_mut(buffers, SecurityBufferType::Stream)?;
-        if buffer.data().len() < decrypted.len() {
+/// 
+/// Fix me !Important: This function does not yet behave the same way as the Windows SSPI.
```

Comment:
nit: "the Windows SSPI" should be just "Windows SSPI"

Reply:
updated